src/main/java/school/faang/bjs245128/Main.java

```diff
@@ -0,0 +1,68 @@
+package school.faang.bjs245128;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+public class Main {
+
+    public static HashSet<Product> products = new HashSet<>();
+
+    public static void main(String[] args) {
+
+    }
+
+    public static void addItem(String category, String name) {
+        products.add(createProduct(category, name));
+    }
+
+    public static String removeItem(String category, String name) {
+        return products.remove(createProduct(category, name)) ? "Product removed from stock" :
+                "Fail. Product does not exists in stock";
+    }
+
+    public static void printAllItems() {
+        products.forEach(System.out::println);
+    }
+
+    public static void findItemsByCategory(String category) {
+        List<Product> filteredProducts = products.stream()
+                .filter(item -> item.getCategory().equals(category))
+                .toList();
+        if (filteredProducts.isEmpty()) {
+            System.out.println("There are no items in stock with this category.");
+        } else {
+            System.out.println("Products in category '" + category + "':");
+            filteredProducts.forEach(System.out::println);
+        }
+    }
+
+    public static HashMap<String, List<Product>> groupProductsByCategory(HashSet<Product> products) {
+        HashMap<String, List<Product>> result = new HashMap<>();
```

Comment:
Ну и небольшой оффтоп. Это делать необязательно, просто для общего развития. Этот метод можно реализовать одной строчкой через стримы (да, они будут изучаться чуть позже, но тем не менее):
`
return products.stream().collect(Collectors.groupingBy(Product::getName));
`
Данный коллектор у стримов как раз таки сгруппирует их в мапу по переданному ключу

Reply:
Yes I know, i deliberately wrote more explanatory rather than with streams. Because streams sometimes hard to grasp from the first sight. Even at work I prefer something in middle( not extreme streams and not cumbersome for loops)